tests/integration/docarray_v2/test_issues.py

```diff
@@ -92,3 +94,50 @@ def foo(self, docs: DocList[A], **kwargs) -> DocList[A]:
     f = Flow().add(uses=MyIssue6084Exec).add(uses=MyIssue6084Exec)
     with f:
         pass
+
+
+@pytest.mark.asyncio
+async def test_issue_6090():
+    """Tests if streaming works with pydantic models with complex fields which are not
+    str, int, or float.
+    """
+
+    class NestedFieldSchema(BaseDoc):
+        name: str = "test_name"
+        dict_field: Dict = Field(default_factory=dict)
+
+    class InputWithComplexFields(BaseDoc):
+        text: str = "test"
+        nested_field: NestedFieldSchema = Field(default_factory=NestedFieldSchema)
+        dict_field: Dict = Field(default_factory=dict)
+        bool_field: bool = False
+
+    class MyExecutor(Executor):
+        @requests(on="/stream")
+        async def stream(
+            self, doc: InputWithComplexFields, parameters: Optional[Dict] = None, **kwargs
+        ) -> InputWithComplexFields:
+            for i in range(4):
+                yield InputWithComplexFields(text=f"hello world {doc.text} {i}")
+
+    docs = []
+    protocol = "http"
+    with Deployment(uses=MyExecutor, protocol=protocol, port=11112) as dep:
```

Comment:
for the tests we have a helper method that gives us a random port, please use it instead of giving a fixed one. That method makes sure there are no collisions